api/pkg/s3/objecthead.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	//"strconv"
+	//"strings"
+
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+)
+
+//ObjectGet -
+func (s *APIService) ObjectHead(request *restful.Request, response *restful.Response) {
+	bucketName := request.PathParameter("bucketName")
+	objectKey := request.PathParameter("objectKey")
+	ctx := context.Background()
+
+	log.Logf("received request for HEAD Object: %s", objectKey)
+
+	objectInput := s3.GetObjectInput{Bucket: bucketName, Key: objectKey}
+
+	objectMD, _ := s.s3Client.GetObject(ctx, &objectInput)
```

Comment:
why get the whole object, when we just need the meta data, can the below example be used ?
`// To retrieve metadata of an object without returning the object itself
//
// The following example retrieves an object metadata.
func ExampleS3_HeadObject_shared00() {
	svc := s3.New(session.New())
	input := &s3.HeadObjectInput{
		Bucket: aws.String("examplebucket"),
		Key:    aws.String("HappyFace.jpg"),
	}

	result, err := svc.HeadObject(input)
	if err != nil {
		if aerr, ok := err.(awserr.Error); ok {
			switch aerr.Code() {
			default:
				fmt.Println(aerr.Error())
			}
		} else {
			// Print the error, cast err to awserr.Error to get the Code and
			// Message from an error.
			fmt.Println(err.Error())
		}
		return
	}

	fmt.Println(result)
}`

Reply:
from [https://github.com/aws/aws-sdk-go/blob/master/service/s3/examples_test.go](url)